edx_django_utils/monitoring/middleware.py

```diff
@@ -60,8 +60,19 @@ def accumulate_metric(cls, name, value):
         Accumulate a custom metric (name and value) in the metrics cache.
         """
         metrics_cache = cls._get_metrics_cache()
-        metrics_cache.setdefault(name, 0)
-        metrics_cache.set(name, value)
+        cached_response = metrics_cache.get_cached_response(name)
+        if cached_response.is_found:
+            try:
+                accumulated_value = value + cached_response.value
+            except TypeError:
+                _set_custom_attribute(
+                    'error_adding_accumulated_metric',
+                    'new_value={}, cached_value={}'.format(value, cached_response.value)
```

Comment:
If the main point of this message is to describe type errors, maybe the arguments should be wrapped in `repr()`?  Otherwise ints and strings could look the same, etc.

Reply:
Added `repr`.

Reply:
@jmbowman: You also made me realize that I forgot to add the `name`, so that has been added as well.

Reply:
I don't see `name` added yet, have you not pushed that commit?

Reply:
Good call.  Just pushed the change.